src/Azure/tests/Unit/AppService/DetectorTest.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Tests\Azure\Unit\AppService;
+
+use OpenTelemetry\Azure\AppService\Detector;
+use OpenTelemetry\SDK\Common\Attribute\Attributes;
+use OpenTelemetry\SemConv\ResourceAttributes;
+use PHPUnit\Framework\TestCase;
+
+class DetectorTest extends TestCase
+{
+    const RESOURCE_GROUP = 'resouce_group';
+    const OWNER_NAME = 'owner_name';
+    const SERVICE_NAME = 'demo-app';
+    const RESOURCE_URI = '/subscriptions/owner_name/resourceGroups/resouce_group/providers/Microsoft.Web/sites/demo-app';
+
+    public function test_valid_app_service_attributes()
+    {
+        $data = [
+            ['azure.app.service.stamp', Detector::ENV_WEBSITE_HOME_STAMPNAME_KEY, 'stamp'],
+            [ResourceAttributes::CLOUD_PLATFORM, null, 'azure_app_service'],
+            [ResourceAttributes::CLOUD_PROVIDER, null, 'azure'],
+            [ResourceAttributes::CLOUD_RESOURCE_ID, Detector::ENV_REGION_NAME_KEY, '/subscriptions/owner_name/resourceGroups/resouce_group/providers/Microsoft.Web/sites/demo-app'],
+            [ResourceAttributes::CLOUD_REGION, Detector::ENV_REGION_NAME_KEY, 'westus'],
+            [ResourceAttributes::DEPLOYMENT_ENVIRONMENT, Detector::ENV_WEBSITE_SLOT_NAME_KEY, 'testing'],
+            [ResourceAttributes::HOST_ID, Detector::ENV_WEBSITE_HOSTNAME_KEY, 'example.com'],
+            [ResourceAttributes::SERVICE_INSTANCE_ID, Detector::ENV_WEBSITE_INSTANCE_ID_KEY, uniqid()],
+            [ResourceAttributes::SERVICE_NAME, Detector::ENV_WEBSITE_SITE_NAME_KEY, self::SERVICE_NAME],
+            [null, Detector::ENV_WEBSITE_RESOURCE_GROUP_KEY, self::RESOURCE_GROUP],
+            [null, Detector::ENV_WEBSITE_OWNER_NAME_KEY, self::OWNER_NAME],
+        ];
+        $expected = [];
+
+        foreach ($data as $item) {
```

Comment:
in the main otel repo, we use assertwell/phpunit-global-state to set/restore environment for tests, eg https://github.com/open-telemetry/opentelemetry-php/blob/main/tests/Unit/SDK/SdkTest.php#L26